### PR TITLE
Fix some BattlePet Opcodes

### PR DESCRIPTION
**Pull request by funjoker**
#507 should fix this
#506 
#505

### CODE REVIEW

File: WowPacketParserModule.V8_0_1_27101/Parsers/BattlePetHandler.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+        public static void ReadPetBattleEffectTarget(Packet packet, int[] targetTypeValueMap, params object[] idx)
+        {
+            var type = packet.ReadBits("Type", 4, idx); // enum PetBattleEffectTargetEx
+
+            packet.ResetBitReader();
+
+            packet.ReadByte("Petx", idx);
+
+            switch (targetTypeValueMap[type])
```
Comment (Shauren) [inline WowPacketParserModule.V8_0_1_27101/Parsers/BattlePetHandler.cs]: uhh, this is exactly the same as `switch (type)`

**Comment by funjoker** [comment]
Well yeah.
I should use 6.x handler for that you are right.

File: WowPacketParserModule.V8_0_1_27101/Parsers/BattlePetHandler.cs (hunk 1 above)
Comment (funjoker) [reply WowPacketParserModule.V8_0_1_27101/Parsers/BattlePetHandler.cs]: Gonna fix that

File: WowPacketParserModule.V8_0_1_27101/Parsers/BattlePetHandler.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
+﻿using WowPacketParser.Enums;
+using WowPacketParser.Misc;
+using WowPacketParser.Parsing;
+
+namespace WowPacketParserModule.V8_0_1_27101.Parsers
+{
+    public static class BattlePetHandler
+    {
+        public static void ReadPetBattleEffectTarget(Packet packet, int[] targetTypeValueMap, params object[] idx)
```
Comment (mdX7) [inline WowPacketParserModule.V8_0_1_27101/Parsers/BattlePetHandler.cs]: targetTypeValueMap unused

File: WowPacketParserModule.V8_0_1_27101/Parsers/BattlePetHandler.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            packet.ReadUInt16("Flags", idx);
+            packet.ReadUInt16("SourceAuraInstanceID", idx);
+            packet.ReadUInt16("TurnInstanceID", idx);
+            packet.ReadSByte("PetBattleEffectType", idx);
+            packet.ReadByte("CasterPBOID", idx);
+            packet.ReadByte("StackDepth", idx);
+
+            var targetsCount = packet.ReadInt32("TargetsCount", idx);
+
+            var targetTypeValueMap = _petBattleEffectTargets;
```
Comment (mdX7) [inline WowPacketParserModule.V8_0_1_27101/Parsers/BattlePetHandler.cs]: _petBattleEffectTargets undefined

File: WowPacketParserModule.V8_0_1_27101/Parsers/BattlePetHandler.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+using WowPacketParser.Misc;
+using WowPacketParser.Parsing;
+
+namespace WowPacketParserModule.V8_0_1_27101.Parsers
+{
+    public static class BattlePetHandler
+    {
+        public static void ReadPetBattleEffectTarget(Packet packet, params object[] idx)
+        {
+            var type = packet.ReadBits("Type", 4, idx); // enum PetBattleEffectTargetEx
```
Comment (mdX7) [inline WowPacketParserModule.V8_0_1_27101/Parsers/BattlePetHandler.cs]: do we have definitions for "PetBattleEffectTargetEx" ?

Comment (funjoker) [reply WowPacketParserModule.V8_0_1_27101/Parsers/BattlePetHandler.cs]: dont think so

**Pull request merged**